course6/toyapp-adam/math/vector.h

```diff
@@ -69,6 +73,7 @@ class Vector
 		return *this;
 	}
 
+	template<typename FloatType = float, typename = IsFloatingPoint<FloatType>>
```

Comment:
There is little point in giving the template param a default value. In most use-cases it will be deduced from call parameters.